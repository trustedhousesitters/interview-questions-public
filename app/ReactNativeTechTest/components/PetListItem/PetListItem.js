import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';

const PetListItem = ({pet, openPopup}) => {
  return (
    <TouchableOpacity style={styles.container} onPress={openPopup}>
      <View style={styles.iconContainer} />
      <View style={styles.nameAndTypeContainer}>
        <Text style={styles.nameText}>{pet.name}</Text>
        <Text style={{color: Colors.dark}}>{pet.type}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 12,
    height: 92,
    flexDirection: 'row',
    alignItems: 'center',
  },
  iconContainer: {
    height: 46,
    width: 46,
    borderRadius: 45,
    backgroundColor: 'grey',
    marginRight: 16,
  },
  nameAndTypeContainer: {
    flex: 1,
  },
  nameText: {
    fontSize: 26,
    fontWeight: 'bold',
    color: Colors.dark,
  },
});

export default PetListItem;
