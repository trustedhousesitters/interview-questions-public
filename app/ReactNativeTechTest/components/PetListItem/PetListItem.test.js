/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import PetListItem from './PetListItem';

describe('Components - PetListItem', () => {
  it('renders - light mode', () => {
    const tree = renderer
      .create(
        <PetListItem
          pet={{
            name: 'Bowser',
            type: 'Rottweiler',
            feeds: 4,
            id: 1,
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
