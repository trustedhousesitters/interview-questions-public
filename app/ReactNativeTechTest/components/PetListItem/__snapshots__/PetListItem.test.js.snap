// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - PetListItem renders - light mode 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "height": 92,
        "marginHorizontal": 12,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "grey",
          "borderRadius": 45,
          "height": 46,
          "marginRight": 16,
          "width": 46,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#444",
            "fontSize": 26,
            "fontWeight": "bold",
          }
        }
      >
        Bowser
      </Text>
      <Text
        style={
          Object {
            "color": "#444",
          }
        }
      >
        Rottweiler
      </Text>
    </View>
  </View>
</View>
`;
